src/actions/addMulticursor.ts

```diff
@@ -22,6 +23,8 @@ const addMulticursor = (state: State, { path, ignoreCursor }: { path: Path; igno
         // on desktop, add the cursor to the multicursor set if it's empty
         ...(isEmpty && !ignoreCursor && state.cursor && !isTouch ? { [hashPath(state.cursor)]: state.cursor } : {}),
       },
+      // update expanded state to include both cursor and new multicursor path
+      expanded: expandThoughts(state, path),
```

Comment:
`state` is the old state before the new multicursor is added. You need to use the most up-to-date state. I recommend adding a new reducer to the flow after `multicursors` is updated.